analyses/eth/notebooks/exploration/eth_ndvi.md

```diff
@@ -0,0 +1,205 @@
+
+Based on the above we suggest to use the mean or median. Due to relatively high fluctuations, I suggest to use the median.
+
+
+#### Adm3 vs Adm2
+It was requested to aggregate the data to adm3. We quickly inspect how they look compared to the admin2's.We can see that there are more than 1000 admin3's. However, due to the high resolution of the data (what is the exact resolution?) and the large size of the country, we still expect it to be okay to aggregate to admin3 if needed.  
+
+```python
+print(f"Number of adm3s: {len(gdf_adm3)}")
+print(f"Number of adm2s: {len(gdf_adm2)}")
+```
+
+```python
+gdf_adm3.boundary.plot()
+```
+
+```python
+gdf_adm2.boundary.plot()
+```
+
+### Define functions
+
+```python
+def aggregate_admin(da,gdf,pcode_col,bins=None):
+    da_clip = da.rio.clip(gdf.geometry, drop=True, from_disk=True)
+    df_stats=compute_raster_statistics(
+        gdf=gdf,
+        bound_col=pcode_col,
+        raster_array=da_clip,
+        lon_coord="x",
+        lat_coord="y",
+        stats_list=["min","median","mean","max","std","count"],
+        #computes value where 20% of the area is above that value
+        percentile_list=[80],
+        all_touched=False,
+    )
+    #find better way to do this
+    #dont understand why but df_stats_adm2.convert_dtypes() is not working
+    df_stats[f"mean_{pcode_col}"]=df_stats[f"mean_{pcode_col}"].astype("float64")
+    df_stats[f"median_{pcode_col}"]=df_stats[f"median_{pcode_col}"].astype("float64")
+    if bins is not None: 
+        df_stats["mean_binned"]=pd.cut(df_stats[f"mean_{pcode_col}"],bins)
+        df_stats["median_binned"]=pd.cut(df_stats[f"median_{pcode_col}"],bins)
+    gdf_stats=gdf[[pcode_col,"geometry"]].merge(df_stats,on=pcode_col)
+    return gdf_stats
+```
+
+```python
+def plt_ndvi_dates(gdf_stats,data_col,colp_num=3):
+    num_plots = len(gdf_stats.date.unique())
+    if num_plots==1:
+        colp_num=1
+    rows = math.ceil(num_plots / colp_num)
+    position = range(1, num_plots + 1)
+    fig=plt.figure(figsize=(30,10))
+    for i,d in enumerate(gdf_stats.date.unique()):
+        ax = fig.add_subplot(rows,colp_num,i+1)
+        gdf_stats[gdf_stats.date==d].plot(ax=ax, column=data_col,
+                             legend=True,# if i==num_plots-1 else False,
+                cmap=ListedColormap(ndvi_colors)#,levels=ndvi_bins
+         )
+        ax.set_title(f"{data_col} of % NDVI of median for {pd.to_datetime(str(d)).strftime('%Y-%m-%d')}")
+        ax.axis("off")
+```
+
+#### Aggregated to admin3
+Below the values per admin3 are shown. We use the same bins as [those used by USGS/FewsNet](https://earlywarning.usgs.gov/fews/product/448). 
+We can see the same pattern as we saw with the raw data, which is a good sign. The dissimilarity between the pattern beginning of December and beginning of January is even more clear from these plots. It remains an open question which we should use.. 
```

Comment:
So you are deciding to just use a single month's plots? I quickly checked the [FewsNet seasonal calendar](https://fews.net/file/113527), looking at the northern/southern pastoral regions, could it be that the north was greener in November/December based on their September/October rains and the November/December southern pastoral rains are what is driving the change in January? Might not make sense to expect patterns to remain the same given how large the country is.

Can you use both maps and also have a map of the differences? This could be January to December differences to see the trend more clearly between the two, rather than the reader having to manually compare. You could also do month-to-month differences of all 3 if you wanted, or December/January differences with November, really depends on the narrative I guess.

Reply:
Yes also agree! Initially the request was just the NDVI right now but I thought that might not reflect the drought situation well as as you note some parts of the country already have their rains in Sep/Oct/Nov. And this if there was extremely low NDVI during that time I would think there might be a high chance there is still significant drought impacts.. I guess that is what I am really struggling with, whether the NDVI at one point gives a good indication of drought.. Especially in a country with different precipitation patterns

Reply:
I think there's so many other factors, don't say anything. I would express these concerns explicitly, link to FewsNet ([is this the latest](https://fews.net/east-africa/ethiopia/food-security-outlook/october-2021)?), and make it clear what you are and aren't saying. Hopefully in country they can interpret the maps in a meaningful/useful way, but yeah, don't think we can really say much more without a lot of additional analysis and exploration.